### DISCUSSION TITLE
Unexpected Attribute Count behavior

### DISCUSSION THREAD

**Discussion by AnakinRaW** [dotnet/runtime#53016] [General]
Hey all, 

I have the following Code:
```cs
[AttributeUsage(AttributeTargets.All, Inherited = true, AllowMultiple = true)]
public class TestAttributeInheritMultiple : Attribute
{
}

public class TestAttributeInheritMultipleDerived : TestAttributeInheritMultiple
{
}

[TestAttributeInheritMultipleDerived]
[TestAttributeInheritMultipleDerived]
public class AttributeTestClass
{
}

[TestAttributeInheritMultipleDerived]
public class AttributeTestClassDerived : AttributeTestClass
{
}
```

when i call `typeof(AttributeTestClassDerived).GetCustomAttributes(true).Length;` i get only **`1`** attribute, though i was expecting the get `3`. 

There are some side observations that lead to my confusion: 
- inspecting the attribute usage of `TestAttributeInheritMultipleDerived ` yields *`Inherited = true and AllowMultiple = true`*. (as expected)
- adding `[AttributeUsage(AttributeTargets.All, Inherited = true, AllowMultiple = true)]` line to `TestAttributeInheritMultipleDerived ` (which should be unnecessary because it already has these usage attributes from its base class) i get **`3`** attributes back for AttributeTestClassDerived. 

Is this expected behavior and I'm overseeing something here, or this is a bug?

Cheers

**Comment by nibdev** [comment]


**Reply by gurustron** [reply]
@nibdev  this discussion is from 2021. It was created waaaaaaay earlier then your question.

**Reply by nibdev** [reply]
@gurustron i'm really sorry, i havent checked the date and was not able to find this discussion or everything else about the topic before...

**Reply by gurustron** [reply]
@nibdev no problem!

**Comment by gurustron** [comment]
Created an issue - https://github.com/dotnet/runtime/issues/88473 in hope that it will gain more attention.
